src/core/core.ts

```diff
@@ -27,7 +28,10 @@ export class Core {
 		return this;
 	}
 
-	generate(schema: Zod.ZodTypeAny, context: Context = { path: [] }): unknown {
+	generate<ZSchema extends ZodTypeAny>(
+		schema: ZSchema,
+		context: Context = { path: [] }
+	): z.infer<typeof schema> {
```

Comment:
Is there a reason why the return type was `unknown` here?

Reply:
Yes. Because the core is not opinionated about what kind of generator you're running, we don't know that generate will result in a schema that matches what is inferred from Zod.

```ts
const SerializedNumber = Generator({
    schema: ZodNumber,
    matches: () => true,
    output: () => ({ type: 'number' })
});

const core = new Core().register([ SerializedNumber ]);
const result = core.generate( z.number() )

// result is not type number, result is { type: 'number' }
```

We also can't currently infer the type statically, because `matches` evaluates at runtime.

BUT `createFixture` (which uses the default generators) DOES produce a matching schema. So we could essentially say that createFixture with no { extend: [] } will produce the exact same type that z.infer dictates.

Reply:
That's what I thought.
I do think in most of the cases the output will match the input schema.
I also expect that most custom generators will also match the schema.
Because of these reasons I changed this behavior, because in 99% (?) of the cases the returntype will be the same as the schema definition.

Reply:
Would it be reasonable to assume that 99% of cases would also be using `createFixture`, so core retains flexibility (for additional use cases) but `createFixture` gives users the convenience of knowing the resulting type?

Reply:
Fair point.
I'll revert these changes, thanks for the reviews!